xdsl/frontend/python_code_check.py

```diff
@@ -298,6 +325,7 @@ def visit_FunctionDef(self, node: ast.FunctionDef) -> ast.FunctionDef:
 
     def visit_Name(self, node: ast.Name) -> ast.Name:
         if node.id == self.name:
+            # TODO: @George what should the type be?
```

Comment:
@georgemitenkov  ?

Reply:
Feels like this might be an actual bug, although I'm probably misunderstanding something

Reply:
What type are you referring to here? This just compares if the should replace the identifier we are visiting (if not, return the old node)

Reply:
It might be easier if you check out the branch and run the pyright locally. The types of the python objects don't match

Reply:
So, this visitor returns a new AST node which replaces the old one, and we want to replace the name with evaluated constant. I added a fix above.